Clarification about OAuth (#292)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

In simpler language, a SCITT Statement could be some vendor-specific software bill of materials (SBOM), results from a model checker, static analyzer, or RATS Evidence about the authenticity of an SBOM creation process, where the Issuer identifies themselves using the `iss` Claim, and the specific software that was analyzed as the Subject using the `sub` Claim.

In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer".
This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens.
This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service.

> Note that the use of {{RFC7523}} is only one possible approach for client authentication in OAuth.

Reviewer: ```suggestion
Highlighting {{RFC7523}} is intended as an example for client authentication originating from OAuth. Many other approaches are applicable.
```

Author: "Many other approaches are applicable." --> The reader will ask himself/herself "which other approaches"

Author: The example of OAuth is actually not so useful as you think. The JWT that is used for client authentication is really only valuable if the JWT has previously been obtained in a secure way. Just creating the JWT on the fly does not provide any useful authentication -- for obvious reason. 

In that sense, the JWT for client authentication in OAuth should be seen as a substitute for a X.509 certificate

In simpler language, a SCITT Statement could be some vendor-specific software bill of materials (SBOM), results from a model checker, static analyzer, or RATS Evidence about the authenticity of an SBOM creation process, where the Issuer identifies themselves using the `iss` Claim, and the specific software that was analyzed as the Subject using the `sub` Claim.

> In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens. This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service. Note that the use of {{RFC7523}} is only one possible approach for client authentication in OAuth.

Reviewer 2: ```suggestion
In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens.
This mirrors how Signed Statements are verified before Receipts are issued by a Transparency Service.
Note that the use of {{RFC7523}} is only one possible approach for client authentication in OAuth.
```

In simpler language, a SCITT Statement could be some vendor-specific software bill of materials (SBOM), results from a model checker, static analyzer, or RATS Evidence about the authenticity of an SBOM creation process, where the Issuer identifies themselves using the `iss` Claim, and the specific software that was analyzed as the Subject using the `sub` Claim.

> In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer". This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens.

Reviewer 2: ```suggestion
In {{RFC7523}}, the Authorization Server (AS) verifies Private Key JWT client authentication requests, and issues access tokens to clients configured to use "urn:ietf:params:oauth:client-assertion-type:jwt-bearer".
This means the AS initially acts as a "verifier" of the authentication credentials in form of a JWT, and then later as an "Issuer" of access and refresh tokens.
```
